M2/Macaulay2/packages/MultiprojectiveVarieties.m2

```diff
@@ -22,11 +22,101 @@ newPackage(
+updatePackageFromDevelopment = method();
+updatePackageFromDevelopment (String,List) := (packagename,listauxfiles) -> (
+    try get "!curl -h 2>&1" else error "please install CURL on your system";
+    <<"-- checking updates for package "<<packagename<<"..."<<endl;
+    dir := temporaryFileName() | "/";
+    mkdir dir;
+    run("curl -s -o "|dir|packagename|".m2 https://raw.githubusercontent.com/Macaulay2/M2/development/M2/Macaulay2/packages/"|packagename|".m2");
+    if not fileExists(dir|packagename|".m2") then error("something went wrong in downloading the package "|packagename);
+    if #listauxfiles > 0 then (
+        makeDirectory(dir|packagename);
+        for f in listauxfiles do (
+            run("curl -s -o "|dir|packagename|"/"|f|".m2 https://raw.githubusercontent.com/Macaulay2/M2/development/M2/Macaulay2/packages/"|packagename|"/"|f|".m2");
+            if not fileExists(dir|packagename|"/"|f|".m2") then error("something went wrong in downloading the package "|packagename);
+        );
+    );
+    get("!"|"cd "|dir|/// && M2 -e "loadPackage(\"///|packagename|///\",FileName=>\"///|dir|packagename|".m2"|///\"); \"///|dir|///packageVersion.m2\"<<///|packagename|///.Options.Version<<close;" 2>&1 &///);
+    v0 := get(dir|"packageVersion.m2");
+    removeFile(dir|"packageVersion.m2");
+    if (value packagename).Options.Version == v0 then (<<"-- package "<<packagename<<" is already updated to version "<<v0<<" -- nothing to do."<<endl; return);
+    e := "";
+    while not(e == "y" or e == "yes" or e == "Y" or e == "Yes") do (
+        e = read("Your version of the package "|packagename|" is outdated. Do you want to download and install the latest version of "|packagename|" now? (y/n) ");
+        if e == "n" or e == "no" or e == "N" or e == "No" then return;
+    );
+    <<"-- installing package "<<packagename<<", version "<<v0<<"... (this may take a while)"<<endl;
+    get("!"|"cd "|dir|/// && M2 -e "uninstallPackage \"///|packagename|///\"" 2>&1 &///);
+    get("!"|"cd "|dir|/// && M2 -e "installPackage(\"///|packagename|///\",FileName=>\"///|dir|packagename|".m2"|///\")" 2>&1 &///);
+    removeFile(dir|packagename|".m2"); 
+    if #listauxfiles > 0 then (
+        for f in listauxfiles do removeFile(dir|packagename|"/"|f|".m2"); 
+        removeDirectory(dir|packagename);  
+    ); 
+    <<"--** package "<<packagename<<", version "<<v0<<", has been successfully installed **--"<<endl;
+    get("!"|///M2 -e "loadPackage \"///|packagename|///\"; \"///|dir|///packageVersion.m2\"<<///|packagename|///.Options.Version<<close;" 2>&1 &///);
+    v := get(dir|"packageVersion.m2");
+    removeFile(dir|"packageVersion.m2");
+    if v != v0 then (
+        sf := (value packagename)#"source file";
+        err := "Error in loading the new version of the package "|packagename|". Check that in your PATH you do not have any older version of the package.";
```

Comment:
Change PATH to 'path', as PATH refers to an environment variable, and path refers to a Macaulay2 variable.